pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/OffsetStartStatus.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.kinesis;
+
+public enum OffsetStartStatus {
```

Comment:
Basically we want to have a flag to mark the first segment of a partition. I think adding a flag in the ZK metadata is more general. Without the flag, we can treat the segment with sequence id 0 as the first segment. The reason why we still want to have a flag:
- When consumption failed on segment 0, we might recreate segment 1 which is also the start of the partition
- We can check if the flag exist to determine whether we are migrating from the inclusive offset to exclusive offset for existing tables

Reply:
Not really getting this point. If this is persisted in ZK, it serves the same purpose right?